cli/helpers.ts

```diff
@@ -0,0 +1,216 @@
+  }
+}
+
+export function copyTemplate(srcDir: string, targetDir: string) {
+  // Copy template, ignoring node_modules
+  try {
+    fse.copySync(srcDir, targetDir, {
+      filter: function (path) {
+        return !(path.indexOf("node_modules") > -1);
+      },
+    });
+    log(chalk.cyan(`Template copied to:\n  ${chalk.cyan.bold(targetDir)}`));
+  } catch {
+    throw new Error(
+      chalk.red(`Error copying template from ${srcDir} to ${targetDir}`)
+    );
+  }
+}
+
+export function updateTemplate(
+  appName: string,
+  targetDir: string,
+  ports: Ports
+) {
+  editReadme(appName, targetDir);
+  editPackageJson(appName, targetDir, Package.Frontend, ports);
+  editPackageJson(appName, targetDir, Package.Backend, ports);
+  editPackageJson(appName, targetDir, Package.TopLevel, ports);
+  editEnvFile(targetDir, Package.Frontend, ports);
+  editEnvFile(targetDir, Package.Backend, ports);
+  log(
+    chalk.cyan(
+      `Updated template with values: 
+  package name: ${chalk.cyan.bold(appName)}
+  frontend port: ${chalk.cyan.bold(ports.frontend)}
+  backend port: ${chalk.cyan.bold(ports.backend)}
+  database port: ${chalk.cyan.bold(ports.db)}
+`
+    )
+  );
+}
+
+export function editReadme(appName: string, targetDir: string) {
+  const filename = path.join(targetDir, `README.md`);
+
+  try {
+    const readmeContents = `#${appName}\n\nGenerated with create-originate-app`;
+    fs.writeFileSync(filename, readmeContents);
+    log(chalk.blue(`Prepared ${filename}`));
+  } catch (err) {
+    throw new Error(chalk.red(`Error preparing ${filename}\n${err}`));
+  }
+}
+
+export function editPackageJson(
+  appName: string,
+  targetDir: string,
+  packageName: Package,
+  ports: Ports
+) {
+  let filename!: string;
+  try {
+    switch (packageName) {
+      case Package.Frontend: {
+        // Update name package.json to use appName
+        filename = path.join(targetDir, `packages/${packageName}/package.json`);
+        let json = JSON.parse(fs.readFileSync(filename).toString());
+        json.scripts["name"] = `@${appName}/${packageName}`;
+        json.scripts[
+          "start:dev"
+        ] = `concurrently -k 'next dev -p ${ports.frontend}' 'yarn codegen:watch'`;
+        fs.writeFileSync(
+          filename,
+          prettier.format(JSON.stringify(json), {
+            semi: false,
+            parser: "json",
+          })
+        );
+        log(chalk.blue(`Prepared ${filename}`));
+        return;
+      }
+      case Package.Backend: {
+        // Update name and db commands in package.json to use appName
+        filename = path.join(targetDir, `packages/${packageName}/package.json`);
+        let json = JSON.parse(fs.readFileSync(filename).toString());
+        json.name = `@${appName}/${packageName}`;
+        json.scripts[
+          "db:start"
+        ] = `docker start ${appName}-postgres 2>/dev/null || docker run -e POSTGRES_PASSWORD=password -p ${ports.db}:5432 -d --name ${appName}-postgres postgres:latest`;
```

Comment:
This is good for now. When we get around to making the db scripts nicer I want to read a `DB_PORT` variable from the `.env` file, and read the package name from `package.json` instead of duplicating that information here.

One issue to note is that there are valid package names that are not valid docker container names. Package names can contain `@` and `/` which are problems. One strategy is to compute a docker container name by deleting all characters except `[a-z0-9-]` from `appName`. Or we could delete invalid characters from the beginning and end, and replace invalid characters in the middle with dashes or something.